docs/Developers/Design/deploy_procedure_design.md

```diff
@@ -0,0 +1,42 @@
+This document describes the actions that have to be executed successfully for
+monstorak's CRDs to be considered reconciled.
+
+![ActionGraph](workflow.dot.png)
+
+# Pre-requisites
+
+## Cluster monitoring deployed
+This operator expects that cluster monitoring stack is already deployed in
+k8s and all required services like prometheus and its related artifacts
+are already available.
+
+## Storage cluster deployed
+The ceph mixins (alerting rules) are meant for storage, so the operator expects
+the storage stack to be deployed and available already. The operator
+![Rook](https://github.com/rook/rook/) takes care of deployment of ceph cluster.
+
+## Monitoring Enabled For Storage
+The namespace in which storage is deployed, monitoring needs to be enabled for
```

Comment:
These steps are performed by the operator already and is OCP specific case which we should not promote.
Rather we should say that `Prometheus should be able to scrape metrics form Storage Cluster`

Reply:
ack